spec/support/shared_contexts/cli/when_main_cli_handler_setup_context.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+shared_context 'when main cli handler setup' do
+  let(:main_handler) do
+    inst = Codeowners::Cli::Main.new
+
+    inst.options = options
+    inst.instance_variable_set(:@config, config)
+    inst.instance_variable_set(:@checker, checker)
```

Comment:
Seems like it can run without setting the checker.

<details>
  <summary>I did a small experiment to prove the concept and found it.</summary>

```ruby
# Experimental instance variable set calls
Fast.shortcut(:exp_remove_var_hack) do

  require 'fast/experiment'
  ruby_files_from('spec/support/shared_contexts').each do |file|
    experiment('Ruby/RemoveInstanceVarHack') do
      lookup file
      search '(send ... instance_variable_set )'
      edit { |node| remove(node.loc.expression) }
      policy { |new_file| system("bundle exec rspec") }
    end.run
  end
end
```
If you want to try [experiments](https://jonatas.github.io/fast/experiments/) yourself, drop this code in a `Fastfile` in the root of codeowners and try to run `fast .exp_remove_var_hack` in the root folder 😉 
</details>